application/views/qslcard/index.php

```diff
@@ -37,39 +48,111 @@
-            echo '<td style=\'text-align: center\'><button onclick="addQsosToQsl(\''.$qsl->filename.'\')" class="btn btn-sm btn-success">' . __("Add Qsos") . '</button></td>';
-            echo '</tr>';
-        }
+			foreach ($qslarray->result() as $qsl) {
+				echo '<tr>';
+				echo '<td style=\'text-align: center\'>'.str_replace("0", "&Oslash;", $qsl->COL_CALL).'</td>';
+				echo '<td style=\'text-align: center\'>';
+				echo $qsl->COL_SUBMODE == null ? $qsl->COL_MODE : $qsl->COL_SUBMODE;
+				echo '</td>';
+				echo '<td style=\'text-align: center\'>';
+				$timestamp = strtotime($qsl->COL_TIME_ON);
+				echo date($custom_date_format, $timestamp);
+				echo '</td>';
+				echo '<td style=\'text-align: center\'>';
+				$timestamp = strtotime($qsl->COL_TIME_ON);
+				echo date('H:i', $timestamp);
+				echo '</td>';
+				echo '<td style=\'text-align: center\'>';
+				if ($qsl->COL_SAT_NAME != null) {
+					echo $qsl->COL_SAT_NAME;
+				} else {
+					echo strtolower($qsl->COL_BAND);
+				};
+				echo '</td>';
+				echo '<td style=\'text-align: center\'>';
+				$timestamp = strtotime($qsl->COL_QSLRDATE ?? '');
+				echo date($custom_date_format, $timestamp);
+				echo '</td>';
+				echo '<td id="'.$qsl->id.'" style=\'text-align: center\'><button onclick="deleteQsl(\''.$qsl->id.'\')" class="btn btn-sm btn-danger">' . __("Delete") . '</button></td>';
+				echo '<td style=\'text-align: center\'><button onclick="viewQsl(\''.$qsl->filename.'\', \''.$qsl->COL_CALL.'\')" class="btn btn-sm btn-success">' . __("View") . '</button></td>';
+				echo '<td style=\'text-align: center\'><button onclick="addQsosToQsl(\''.$qsl->filename.'\')" class="btn btn-sm btn-success">' . __("Add Qsos") . '</button></td>';
+				echo '</tr>';
+			}
 
-        echo '</tbody></table>';
-    }
-    ?>
+			echo '</tbody></table>';
+		}
+		?>
+	</div>
+
+	<!-- Gallery View -->
+	<div id="galleryView" class="qsl-gallery" style="display: none;">
+		<div class="waterfall-grid">
+			<?php if (is_array($qslarray->result())): ?>
+				<?php foreach ($qslarray->result() as $qsl): ?>
+					<?php
+					$qslDate = null;
+					$timestamp = strtotime($qsl->COL_TIME_ON);
+					if (!empty($qsl->COL_QSLRDATE)) {
+						$qslDate = date($custom_date_format, strtotime($qsl->COL_QSLRDATE));
+					}
+					$band = ($qsl->COL_SAT_NAME != null) ? $qsl->COL_SAT_NAME : strtolower($qsl->COL_BAND);
+					$mode = $qsl->COL_SUBMODE == null ? $qsl->COL_MODE : $qsl->COL_SUBMODE;
+
+					// Extract the user ID and filename for constructing the correct path
+					$parts = explode('/', $qsl->filename);
+					$filename = end($parts);
+					$userdata_dir = $this->config->item('userdata');
+					$user_id = $this->session->userdata('user_id');
+
+					// Build correct image path: userdata/[user_id]/qsl_card/[filename]
+					$image_path = $this->paths->getPathQsl() . '/' . $filename;
```

Comment:
We need to add the base_url here. See https://github.com/Cryolitia-Forks/wavelog/pull/2.

Reply:
done